docs/http_proxy.md

```diff
@@ -0,0 +1,40 @@
+
+HTTP Forward Proxy Support
+==========================
+
+It is possible to launch a kubernetes cluster from behind an http forward proxy ("corporate proxy").  To do so, you will need to configure the `egressProxy` for the cluster.
+
+It is assumed the proxy is already existing.  If you want a private topology on AWS, for example, with an proxy instead of a NAT instance, you'll need to create the proxy yourself.  See [Running in a shared VPC](run_in_existing_vpc.md).
+
+This configuration only manages proxy configurations for Kops and the Kubernetes cluster.  We can not handle proxy configuration for application containers and pods.
+
+## Configuration
+
+Add `spec.egressProxy` port and url as follows
+
+``` yaml
+spec:
+  egressProxy:
+    httpProxy:
+      host: proxy.corp.local
+      port: 3128
+```
+
+Currently we assume the same configuration for http and https traffic.
+
+## Proxy Excludes
+
+Most clients will blindly try to use the proxy to make all calls, even to localhost and the local subnet, unless configured otherwise.  Some basic exclusions necessary for successful launch and operation are added for you at initial cluster creation.  If you wish to add additional exclusions, add or edit `egressProxy.excludes` with a comma separated list of hostnames.  Matching is based on suffix, ie, `corp.local` will match `images.corp.local`, and `.corp.local` will match `corp.local` and `images.corp.local`, following typical `no_proxy` environment variable conventions.
+
+``` yaml
+spec:
+  egressProxy:
+    httpProxy:
+      host: proxy.corp.local
+      port: 3128
+    excludes: corp.local,internal.corp.com
```

Comment:
So this is only going to work for layer 7 proxies, and if we - for example - but a NAT gateway in here it would not work as expected.  That said I think this is fine, it can be a validation error.  YAGNI :-)